### PR TITLE
#472: add eslint + fix eslint error on server API + update doc

### PR DESCRIPTION
**Pull request by BernardJeremy**

### CODE REVIEW

File: websocket/daemon/software.js [hunk 1]
```diff
@@ -40,8 +40,8 @@ module.exports.removePackages = function (userName, packages, onStatusChange) {
   return exec(userName, 'remove_packages', packages, onStatusChange);
 };
 
-module.exports.queryPackage = function (userName, package, onStatusChange) {
```
Comment (Grenadingue) [inline websocket/daemon/software.js:43]: Why updating the variables name ?

File: managers/users.js [hunk 2]
```diff
@@ -665,7 +665,7 @@ module.exports.deleteUsers = function () {
   };
 };
 
-function retrieveUserFromId(id) {
```
Comment (Grenadingue) [inline managers/users.js:668]: Why adding `req` ?

File: managers/software.js [hunk 3]
```diff
@@ -33,8 +33,8 @@ module.exports.allUsersInfo = function (socket) {
 module.exports.allSoftwaresByUser = function (user, socket) {
   softwareAdapter.launchListPackages(user)
     .then(function (listpackage) {
-      listpackage.result.forEach (function (package){
```
Comment (Grenadingue) [inline managers/software.js]: Why updating the variables name ?

Comment (BernardJeremy) [reply managers/software.js]: `package` is a reserved variable name in JS

File: websocket/daemon/software.js (hunk 1 above)
Comment (BernardJeremy) [reply websocket/daemon/software.js:43]: `package` is a reserved variable name in JS

File: managers/users.js (hunk 2 above)
Comment (BernardJeremy) [reply managers/users.js:668]: Because you use it in this function ;)

Comment (Grenadingue) [reply managers/users.js:668]: Oh god, so silly mistakes...

**Review by Grenadingue — APPROVED** [review]

**Pull request merged**